omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.web.filter;
+
+import static org.openmrs.web.WebConstants.OPENMRS_LOGIN_REDIRECT_HTTPSESSION_ATTR;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.api.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This filter serves to set the redirect url on the session, if any GET request is received for an
+ * unauthenticated user, so that any downstream redirection after login can bring the user to the
+ * page they had been requesting.
+ */
+public class RedirectAfterLoginFilter implements Filter {
+	
+	private static final Logger log = LoggerFactory.getLogger(RedirectAfterLoginFilter.class);
+	
+	private FilterConfig config;
+	
+	/**
+	 * @see Filter#doFilter(ServletRequest, ServletResponse, FilterChain)
+	 */
+	public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
+		try {
+			HttpServletRequest httpReq = (HttpServletRequest) req;
+			String requestURI = httpReq.getRequestURI();
+			if (!Context.isAuthenticated()) {
+				if ("GET".equalsIgnoreCase(httpReq.getMethod()) && !requestURI.contains("login.")) {
+					if (req.getAttribute(OPENMRS_LOGIN_REDIRECT_HTTPSESSION_ATTR) == null) {
```

Comment:
Does this check actually work? I can see we use it elsewhere, but it seems like this should be something like:

```java
req.getSession(false).getAttribute(OPENMRS_LOGIN_REDIRECT_HTTPSESSION_ATTR) == null
```

When I try naïvely to call:

```java
req.getSession().setAttribute("MyAttribute", "SomeValue");
```

And then do:

```java
req.getAttribute("MyAttribute")
```

The response is still `null`.

Reply:
Oops, sorry, yeah, good catch.  I was initially setting a request attribute and then changed it when I recognized it was supposed to be a session attribute, but didn't change this.